features/all-positions/LiquidationActionDialog.tsx

```diff
@@ -0,0 +1,488 @@
+import styled from "styled-components";
+import { utils } from "ethers";
+const { formatUnits: fromWei, parseUnits: toWei } = utils;
+import { useState } from "react";
+
+import {
+  Box,
+  Container,
+  Typography,
+  Dialog,
+  Button,
+  Tooltip,
+} from "@material-ui/core";
+
+import EmpState from "../../containers/EmpState";
+import DvmState from "../../containers/DvmState";
+import Token from "../../containers/Token";
+import EmpSponsors from "../../containers/EmpSponsors";
+import EmpContract from "../../containers/EmpContract";
+import Collateral from "../../containers/Collateral";
+import Etherscan from "../../containers/Etherscan";
+
+import { DOCS_MAP } from "../../utils/getDocLinks";
+import { isPricefeedInvertedFromTokenSymbol } from "../../utils/getOffchainPrice";
+
+const Label = styled.span`
+  color: #999999;
+`;
+
+const Status = styled(Typography)`
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+`;
+
+const PositionDialog = styled.div`
+  width: 100%;
+  border-color: white;
+  padding-left: 15px;
+  padding-right: 15px;
+  padding-bottom: 20px;
+`;
+
+const Important = styled(Typography)`
+  color: red;
+  background: black;
+  display: inline-block;
+`;
+
+const Link = styled.a`
+  color: white;
+  font-size: 14px;
+`;
+
+interface DialogProps {
+  handleClose(): any;
+  isDialogShowing: boolean;
+  selectedSponsor: string | null;
+}
+
+const LiquidationActionDialog = (props: DialogProps) => {
+  const { empState } = EmpState.useContainer();
+  const { dvmState } = DvmState.useContainer();
+  const { finalFee } = dvmState;
+  const { getEtherscanUrl } = Etherscan.useContainer();
+  const { contract: emp } = EmpContract.useContainer();
+
+  const { symbol: tokenSymbol } = Token.useContainer();
+  const {
+    symbol: collSymbol,
+    balance: collBalance,
+    allowance: collAllowance,
+    setMaxAllowance: setMaxCollateralAllowance,
+  } = Collateral.useContainer();
+  const { liquidations } = EmpSponsors.useContainer();
+
+  const [hash, setHash] = useState<string | null>(null);
+  const [success, setSuccess] = useState<boolean | null>(null);
+  const [error, setError] = useState<Error | null>(null);
+
+  const {
+    disputeBondPct,
+    priceIdentifier: priceId,
+    liquidationLiveness,
+    withdrawalLiveness,
+    currentTime,
+  } = empState;
+
+  const prettyBalance = (x: number) => {
+    const x_string = x.toFixed(4);
+    return utils.commify(x_string);
+  };
+
+  const prettyAddress = (x: String) => {
+    return x.substr(0, 6) + "..." + x.substr(x.length - 6, x.length);
+  };
+
+  if (
+    emp &&
+    liquidations &&
+    props.selectedSponsor &&
+    liquidations[props.selectedSponsor] &&
+    liquidationLiveness &&
+    priceId &&
+    disputeBondPct &&
+    collAllowance &&
+    collBalance &&
+    collSymbol &&
+    disputeBondPct &&
+    liquidationLiveness &&
+    withdrawalLiveness &&
+    currentTime &&
+    tokenSymbol &&
+    finalFee
+  ) {
```

Comment:
Same as in the other PR -- I think we should check _for_ null and return early if any of these are null.

Reply:
as with the other PR, if any of these are null then nothing should be loaded; we can return early but this pattern I have here is the same as all the other components in the dapp. what do you think?

Reply:
As discussed IRL, this makes sense for the sake of consistency.

For the record I think early returns have two major readability advantages:
1. Less nested logic.
2. You don't have to search to find the else or see if it exists.

I think we should open a quick issue to define the right "pattern" for doing null checks and default renders. But this LGTM!